### ISSUE TITLE
`zenstack_` fields are not stripped from nested relation documents

### ISSUE THREAD

**Issue by ymc9** [zenstackhq/zenstack#386]
Schema:

```prisma
        model User {
            id String @id @unique @default(uuid())
            posts Post[]

            @@allow('all', true)
        }
            
        model Post {
            id String @id @default(uuid())
            title String
            published Boolean @default(false)
            author User @relation(fields: [authorId], references: [id])
            authorId String
        
            @@allow('all', true)
        }   
```

Query:
```ts
withPolicy(prsima).user.findMany({ include: { posts: true } });
```

Result:
The nested `posts` documents have `zenstack_` fields.

**Comment by ymc9** [comment]
Reported by @sidharthv96

**Comment by ymc9** [comment]
Fixed by #400
